src/main/java/com/battleship/backend/models/Ship.java

```diff
@@ -25,7 +29,7 @@ public Ship(String name, int length) {
 
     public void fillShip(){
         for(int i = 0; i < shipSections.length; i++) {
-            shipSections[i] = new ShipSection(this);
+            shipSections[i] = new ShipSection(getName());
         }
```

Comment:
Do you need to call a getter here or just this.name?

Reply:
I thought using the getter was preferable? Or is that only outside of the class?

Reply:
Yes you only need the getter from outside the class

Reply:
Reading up a little further, if you had some logic in `getName()` then you would call `getName()` instead of the instance variable.  But, if you don't have any logic and it's just a getter, you can just call the variable.  Either case, it's your preference the approach you want to take.